source/common.ts

```diff
@@ -110,10 +107,44 @@ const i18n = {
   _normalizeWithAncestorsCache: {} as Record<string, readonly string[]>,
   _translations: {} as JSONObject,
   _ts: 0,
+  _interpolateTranslation(
+    variables: Record<string, unknown>,
+    translation: string,
+  ) {
+    let interpolatedTranslation = translation;
+    Object.entries(variables).forEach(([key, value]) => {
+      const tag = i18n.options.open + key + i18n.options.close;
+      if (interpolatedTranslation.includes(tag)) {
+        interpolatedTranslation = interpolatedTranslation
+          .split(tag)
+          .join(value as string);
+      }
+    });
+    return interpolatedTranslation;
+  },
+  _normalizeGetTranslation(locales: string[], key: string) {
+    let finalTranslation: unknown;
+    let translation: unknown;
+    locales.some(locale =>
+      i18n._normalizeWithAncestors(locale).forEach(locale => {
+        translation = get(i18n._translations, `${locale}.${key}`);
+        translation !== undefined && (finalTranslation = translation);
+      }),
+    );
```

Comment:
Something went wrong here - `some` suggests we'll do an early return, but then a `forEach`. Also, we're doing `get` quite a lot of times here (potentially `#locales * #_normalizeWithAncestors`) but use only the last one. This should be simplified.